Metrics/collect_and_upload.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```

Comment:
A few thoughts on this:

  * This script could possibly be in `scripts` along with the rest of them.
  * Needs a copyright notice
  * Should either check errors or have bash do it for you using `set -euo pipefail` near the top

Reply:
All good points. Sorry, I created this PR to play around with how things work on Travis before cleaning up for review.

Will resolve when addressed.

Reply:
Confirmed that we can successfully upload metrics from Travis now. Addressed these comments here too.

Still cleaning up the PR, and will comment when it is ready for review.